scrabble_challenge/scrabble_challenge.py

```diff
@@ -0,0 +1,74 @@
+import sys
+
+CHARACTERS_SCORE = {"a": 1, "c": 3, "b": 3, "e": 1, "d": 2, "g": 2,
+                    "f": 4, "i": 1, "h": 4, "k": 5, "j": 8, "m": 3,
+                    "l": 1, "o": 1, "n": 1, "q": 10, "p": 3, "s": 1,
+                    "r": 1, "u": 1, "t": 1, "w": 4, "v": 4, "y": 4,
+                    "x": 8, "z": 10}
+
+
+def get_scrabble_score(valid_word):
+    """
+    Calculates the scrabble score of valid words and returns it.
+    """
+    scrabble_score = 0
+    for character in valid_word:
+        scrabble_score += CHARACTERS_SCORE[character]
+    return scrabble_score
+
+
+def is_word_from_rack(word, rack):
+    """
+    This function check the word that if it is made of the letters that are subset of rack letters.
+    """
+    rack = str(rack)
+    rack = rack.lower()
+    for character in word:
+        if character not in rack:
+            return False
+        else:
+            rack = rack.replace(character, "", 1)
+    return True
+
+
+def display_score(valid_words_list):
+    """
+    This function takes the list of valid words and then get their score and dispalys final result.
+    """
+    if valid_words_list != []:
+        valid_words_dictionary = {}
+        for word in valid_words_list:
+            scrabble_score_of_valid_word = get_scrabble_score(word)
+            valid_words_dictionary[word] = scrabble_score_of_valid_word
+        result_sorted_by_score = sorted(valid_words_dictionary.items(), key=lambda kv: kv[1], reverse=True)
+        for word, score in result_sorted_by_score:
+            print(str(score) + " " + word)
+    else:
+        print("No valid word is found")
+
+
+def find_valid_words(rack):
+    """
+    This function gives the valid words that are  made of the letters that are subset of rack letters.
+    """
+    file_path = "sowpods.txt"
+    file = open(file_path, "r")

```

Comment:
Use with open to open file, in this way we don't have to worry about closing the file.